### PR TITLE
Sparkline chart

### PR DESCRIPTION
**Pull request by 4p00rv**
Add Sparkline chart for purchase price in trading window.

### CODE REVIEW

File: src/javascript/binary/websocket_pages/trade/price.js [hunk 1]
```diff
@@ -27,8 +28,99 @@ const formatMoney          = require('../../common_functions/currency').formatMo
  * `Price.display()` to display the price details returned from server
  */
 const Price = (() => {
+    
+
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/trade/price.js]: can we remove these two white lines?

File: src/javascript/binary/websocket_pages/trade/price.js [hunk 2]
```diff
@@ -27,8 +28,99 @@ const formatMoney          = require('../../common_functions/currency').formatMo
...
+                overflow: 'visible',
+            },
+        },
+        title: {
+            text: '',
+        },
+        credits: {
+            enabled: false,
+        },
+        exporting: { enabled: false },
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/trade/price.js]: can we make this style consistent? either one variable objects should be written in one line or have line break

File: src/javascript/binary/websocket_pages/trade/price.js [hunk 3]
```diff
@@ -27,8 +28,99 @@ const formatMoney          = require('../../common_functions/currency').formatMo
  * `Price.display()` to display the price details returned from server
  */
 const Price = (() => {
+    
+
     let type_display_id_mapping = {};
     let form_id                 = 0;
+    let Highcharts;
+    const sparkline_options = {
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/trade/price.js]: is it worth putting this config in another file? it doesn't really have anything to do with price.js

File: src/javascript/binary/websocket_pages/trade/price.js [hunk 4]
```diff
@@ -230,7 +325,40 @@ const Price = (() => {
         }
     };
 
+    /**
+     * @param {*} ele : ele which needs to be updated or needs to have a sparkline chart.
+     * @param {*} value : value of point to be added
+     * @param {*} type : type of data being added to chart, can be any arbitrary value.
+     * @param {*} data_points : Maximum number of data points allowed.
+     */
+    const showSparkline = (ele, val, type, data_points = 10) => {
```
Comment (negarn) [inline src/javascript/binary/websocket_pages/trade/price.js]: same here, is it worth putting it with the config file in another file?

File: src/javascript/binary/common_functions/sparkline_chart.js [hunk 5]
```diff
@@ -115,7 +113,8 @@ const showSparkline = (ele, val, data_points = 10) => {
                 name: 'Price',
                 data: [{ y: value }],
             }];
-            Highcharts.chart(ele, sparkline_options);
+            const chart = new Highcharts.chart(ele, sparkline_options);
```
Comment (kellybinary) [inline src/javascript/binary/common_functions/sparkline_chart.js]: is `chart` being used anywhere?

File: src/javascript/binary/common_functions/sparkline_chart.js [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                last_point.update({
+                    marker: { enabled: false },
+                });
+            }
+
+            if (data.length < data_points) {
+                series.addPoint({ y: value }, true);
+            } else {
+                series.addPoint({ y: value }, true, true);
+            }
```
Comment (binary-hamedanchi) [inline src/javascript/binary/common_functions/sparkline_chart.js]: How about replacing these 5 lines with:
```js
series.addPoint({ y: value }, true, data.length >= data_points);
```

File: src/javascript/binary/common_functions/sparkline_chart.js [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+        },
+    },
+    series: [{
+        name: 'Price',
+        data: [],
+    }],
+};
+
+// Initialize Highcharts.
+getHighstock((H) => {
```
Comment (binary-hamedanchi) [inline src/javascript/binary/common_functions/sparkline_chart.js]: This causes to load the highstock js file on all pages. Needs to be called on demand.
Maybe moving it inside the `showSparkline` and use as a promise?

**Comment by 4p00rv** [comment]
@binary-hamedanchi Thanks for reviewing. The requested changes have been implemented.

**Comment by binary-hamedanchi** [comment]
@4p00rv Noted with thanks.

**Pull request closed without merging**